sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTreeUpdated.java

```diff
@@ -0,0 +1,249 @@
+		CssStyles.INDENT_LEFT_2,
+		CssStyles.INDENT_LEFT_3 };
+
+	private Map<ENUM, CheckBox> enumToggles = new HashMap<>();
+	private VerticalLayout checkBoxTreeLayout;
+	private HorizontalLayout checkBoxTreeLayoutForGroups;
+	private Class<ENUM> enumType;
+	private boolean addVerticalSpaces;
+	private UnaryOperator<ENUM> parentProvider;
+	private Function<ENUM, ? extends Enum> groupProvider;
+	private Class<? extends Enum> parentGroup;
+	private boolean settingToggles = false;
+	private int columns;
+
+	public CheckBoxTreeUpdated() {
+
+	}
+
+	public Map<ENUM, CheckBox> getEnumToggles() {
+		return enumToggles;
+	}
+
+	@Override
+	protected Component initContent() {
+
+		if (enumType != null) {
+			if (groupProvider == null) {
+				checkBoxTreeLayout = new VerticalLayout();
+				checkBoxTreeLayout.setMargin(false);
+				checkBoxTreeLayout.setSpacing(false);
+				checkBoxTreeLayout.setWidth(100, Sizeable.Unit.PERCENTAGE);
+				addCheckBoxes();
+				final MarginInfo marginInfo = new MarginInfo(false, false, true, false);
+				checkBoxTreeLayout.setMargin(marginInfo);
+			} else {
+				checkBoxTreeLayoutForGroups = new HorizontalLayout();
+				checkBoxTreeLayoutForGroups.setMargin(false);
+				checkBoxTreeLayoutForGroups.setWidth(100, Unit.PERCENTAGE);
+				addGroups();
+			}
+
+			setToggleValues();
+		}
+
+		addValueChangeListener(valueChangeEvent -> {
+			Map<ENUM, Boolean> value = (Map<ENUM, Boolean>) valueChangeEvent.getProperty().getValue();
+			final Map<ENUM, Boolean> nullSafeValue = value == null ? new HashMap<>() : value;
+			settingToggles = true;
+			enumToggles.forEach(((anEnum, checkBox) -> {
+				final Boolean orDefault = nullSafeValue.getOrDefault(anEnum, false);
+				if (!Objects.equals(checkBox.getValue(), orDefault)) {
+					checkBox.setValue(orDefault);
+				}
+			}));
+			settingToggles = false;
+		});
+
+		if (checkBoxTreeLayout != null) {
+			return checkBoxTreeLayout;
+		}
+
+		return checkBoxTreeLayoutForGroups;
+	}
+
+	private void setToggleValues() {
+		settingToggles = true;
+		setToggleValues(getValue());
+		settingToggles = false;
+	}
+
+	private void addGroups() {
+		final ENUM[] enumElements = enumType.getEnumConstants();
+
+		Map<Enum, List<ENUM>> enumGroupMap = new HashMap<>();
+
+		for (ENUM enumElement : enumElements) {
+			final Enum groupElement = groupProvider.apply(enumElement);
+			if (enumGroupMap.containsKey(groupElement)) {
+				enumGroupMap.get(groupElement).add(enumElement);
+			} else {
+				enumGroupMap.put(groupElement, new ArrayList<>());
+				enumGroupMap.get(groupElement).add(enumElement);
+			}
+		}
+
+		List<VerticalLayout> columnList = new ArrayList<>();
+
+		for (int i = 0; i < columns; i++) {
+			final VerticalLayout e = new VerticalLayout();
+			e.setMargin(false);
+			e.setSpacing(false);
+			columnList.add(e);
+		}
+
+		AtomicInteger currentColumn = new AtomicInteger();
+		currentColumn.set(0);
+
+		Arrays.stream(parentGroup.getEnumConstants()).forEach(enumListEntry -> {
```

Comment:
rewrite to indexed for cycle and avoid using `AtomicInteger currentColumn`

Reply:
rewritten